### ISSUE TITLE
publishPlanResults results in a blank Terraform Plan view

### ISSUE THREAD

**Issue by hallgeirl** [charleszipp/azure-pipelines-tasks-terraform#255]
Hi!
I'm trying to get some output to the terraform plan view.. The view just shows:
```No terraform plans have been published for this pipeline run. The terraform cli task must run plan with publishPlanResults: string (where string represents the plan name) to view plans.```

I have set the `publishPlanResults` input on the plan task:

```
    - task: TerraformCLI@0
      displayName: 'terraform plan'
      inputs:
          command: plan
          environmentServiceName: 'Dev'
          workingDirectory: 'team-dev-spaces/platform/infrastructure/test-infra-deploy/terraform'
          runAzLogin: true
          publishPlanResults: 'testinfra'
          commandOptions: '-out=$(Pipeline.Workspace)/testinfra.tfplan -detailed-exitcode'
```

I also set System.Debug=true and inspected the debug output of the plan command. I could not see any obvious errors. I have attached the file to this issue.
[tfoutput.txt](https://github.com/charleszipp/azure-pipelines-tasks-terraform/files/6220686/tfoutput.txt)

I'm also publishing the plan result as an artifact (not sure if that is needed):
```
    - task: PublishPipelineArtifact@1
      displayName: 'Publish Artifact: terraform-plans'
      inputs:
        targetPath: $(Pipeline.Workspace)/testinfra.tfplan
        artifact: 'terraform-plans'
```

Is there something I am missing? Does the value in the publishPlanResults input need to have a particular value, matched to something else? I couldn't find anything about this in the docs.

I'm using the latest version of the extension (0.6.15), and the latest version of Terraform (0.14.9). The pipeline is running on an Ubuntu 18.04 agent, and I'm using Azure Storage as the backend storage.

Is this a bug, or a user error (me)? :-)

**Comment by charleszipp** [comment]
@hallgeirl can you open the browser dev tools and see if there are any errors appearing in the console after you select the terraform plan tab? If there are errors, can you post the message you see here?

**Comment by alanblythe** [comment]
@charleszipp I had this problem as well and tracked it down. It's caused by use of the old visualstudio.com URLs. If I toggle my org to visualstudio.com the plan tab no longer works. For the same pipeline run I can toggle to dev.azure.com and it starts to work again.
The URL being accessed is as follows:
https://alanblythe.visualstudio.com/_apis/_apis/build/builds/32/0a5062b1-563d-50ca-cfde-4e65f54d0cf7/attachments/attachments/terraform-plan-results/plan-stdappsvc-web-dev
which, results in a 400 with the following message
"message":"TF200016: The following project does not exist: _apis. Verify that the name of the project is correct and that the project exists on the specified Azure DevOps Server."
It looks like it's a simple URL construction issue.
I've also attached an image.
![image](https://user-images.githubusercontent.com/2677911/114616923-743a0c00-9c75-11eb-8d20-f3bca4a6b787.png)

**Comment by charleszipp** [comment]
Thanks for sharing @alanblythe! This then looks to be an issue with the `azure-devops-extension-sdk` and/or `azure-devops-extension-api`. This extension uses those libraries to make the api requests. The host name is configured by those libraries themselves; not this extension. Therefore, it would appear that the way they derive the host name may assume everyone is accessing their orgs via dev.azure.com.

**Comment by OmnipotentOwl** [comment]
@charleszipp, I ran into this issue as well and dug into the code running in Azure DevOps to look at the root cause. One of the things I noticed was that when using the visualstudio.com URL for an organization the code that handles the mapping of the returned href to an AzdoAttachment object in typescript isn't handling the pathing difference in the URL format. I created a branch in my fork of the repo with a workaround for the issue that I think should handle the special case of the legacy URL. 

I've also included a screenshot below of the segments from the visualstudio.com URL.
![VisualStudio url issue](https://user-images.githubusercontent.com/1769881/115052020-72598e00-9eab-11eb-917f-32ff671b3c76.PNG)
